tests/testthat/test-list-combine.R

```diff
@@ -70,6 +70,24 @@ test_that("NULLs are ignored", {
   expect_equal(list_cbind(list(df1, NULL, df2)), vec_cbind(df1, df2))
 })
 
+test_that("NULLs are converted to NA when keep_empty = TRUE", {
+  df1 <- data.frame(x = 1)
+  df2 <- data.frame(y = 1)
+
+  expect_equal(
+    list_c(list(1, NULL, 2), keep_empty = TRUE),
+    c(1, NA, 2)
+  )
+  expect_equal(
+    list_rbind(list(df1, NULL, df1), keep_empty = TRUE),
+    data.frame(x = c(1, NA, 1))
+  )
+  expect_equal(
+    list_cbind(list(df1, z = NULL, df2), keep_empty = TRUE),
+    data.frame(df1, z = NA, df2)
+  )
+})
```

Comment:
If we decide to implement this then I'd like the chance to come through and write _a lot_ of tests for the 3 cases individually. The `keep_empty` logic in tidyr is quite hard to get 100% right, and required a lot of edge case tests.